File: RookiesFashion.ClientSite/Models/Product.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace RookiesFashion.ClientSite.Models
{
    public class Product
    {
        public int ProductId { get; set; }
        public string? Name { get; set; }
        public string? Description { get; set; }
        public double? Price { get; set; }
        public int? CategoryId { get; set; }
        public DateTime CreatedDate { get; set; }
        public Category? Category { get; set; }
        public List<Rating>? Ratings { get; set; }
        public double? AvgRating { get; set; }
        public IEnumerable<UpdatedDate>? UpdatedDates { get; set; }
        public IEnumerable<Image>? Thumbnail { get; set; }
        public IEnumerable<Color>? Colors { get; set; }
        public IEnumerable<Size>? Sizes { get; set; }
    }
}

File: rookies-fashion-admin-site/src/types/model/Product.ts
import {
  Category,
  Rating,
  Image,
  Color,
  Size,
  UpdatedDate,
} from ".";

export type Product = {
  ProductId: number;
  Name: string;
  Description: string;
  Price: number;
  CategoryId: number;
  CreatedDate: string;
  Category: Category;
  Ratings: Array<Rating>;
  AvgRating: number;
  Thumbnail: Array<Image>;
  UpdatedDates: Array<UpdatedDate> | undefined;
  Colors: Array<Color>;
  Sizes: Array<Size>;
}
